src/watch.rs

```diff
@@ -16,6 +16,40 @@ pub struct Watch {
     watches: HashSet<PathBuf>,
 }
 
+/// Count of extra events batched with this change.
+#[derive(Clone, Debug)]
+pub struct NumberOfBatchedEvents(u16);
+
+/// A debug message string that can only be displayed via `Debug`.
+#[derive(Clone, Debug)]
+pub struct DebugMessage(String);
+
+impl From<String> for DebugMessage {
+    fn from(s: String) -> Self {
+        DebugMessage(s)
+    }
+}
+
+/// Description of the project change that triggered a build.
+#[derive(Clone, Debug)]
+pub enum Reason {
+    /// When a project is presented to Lorri to track, it's built for this reason.
+    ProjectAdded(PathBuf),
+    /// When there is a filesystem change, the first changed file is recorded,
+    /// along with a count of other filesystem events.
+    FilesChanged(PathBuf, NumberOfBatchedEvents),
```

Comment:
Why only a single PathBuf, instead of a list of all path changes we saw?

Reply:
My experience with inotify has been that events come in big and strange batches. I'd wanted to try it out for a little while before unwrapping "and 3 other files" into the list., in case it was often "and 300 other files." If that's not a worry (i.e. we're okay with 300 lines in log files if that happens), that's easy enough to change to.